backend/src/org/commcare/xml/DetailFieldParser.java

```diff
@@ -84,16 +84,15 @@ public DetailField parse() throws InvalidStructureException, IOException, XmlPul
         return builder.build();
     }
 
-    private void parseStyle(DetailField.Builder builder) throws InvalidStructureException, IOException, XmlPullParserException {
-        //style
-        if (parser.getName().toLowerCase().equals("style")) {
-            StyleParser styleParser = new StyleParser(builder, parser);
-            styleParser.parse();
-            //Header
+    private void parseGridStyle(DetailField.Builder builder) throws InvalidStructureException, IOException, XmlPullParserException {
+        if (parser.getName().toLowerCase().equals("style") ||
+                parser.getName().toLowerCase().equals("grid-style")) {
```

Comment:
updating the parser to accept either `style` or `grid-style` as the element name